sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -658,24 +721,151 @@ class GroupedAggregations extends Arm {
       (groupBy, aggs) => groupBy.aggregateWindowsOverRanges(aggs: _*))
   }
 
+  private final def doRunningWindowScan(
+      inputCb: ColumnarBatch,
+      outputColumns: Array[ColumnVector]): Unit = {
+    runningWindowOptimizedData.foreach {
+      case (func, outputIndexes) =>
+        val agg = func.scanAggregation
+        val replace = func.scanReplaceNulls
+        require(func.boundInputLocations.length == 1)
+        val inputColIndex = func.boundInputLocations.head
+        val inputCol = inputCb.column(inputColIndex).asInstanceOf[GpuColumnVector].getBase
+        val replaced = withResource(inputCol.scan(agg, ScanType.INCLUSIVE, NullPolicy.EXCLUDE)) {
+          scanned =>
+            // For scans when nulls are excluded then each input row that has a null in it the
+            // output row also has a null in it. Typically this is not what we want, because
+            // for windows that only happens if the first values are nulls. So we will then call
+            // replace nulls as needed to fix that up. Typically the replacement policy is
+            // preceding.
+            if (replace.isDefined) {
+              scanned.replaceNulls(replace.get)
+            } else {
+              scanned.incRefCount()
+            }
+        }
+
+        withResource(replaced) { replaced =>
+          withResource(castIfNeeded(replaced, func.dataType)) { retCol =>
+            outputIndexes.foreach { outIndex =>
+              outputColumns(outIndex) = retCol.incRefCount()
+            }
+          }
+        }
+    }
+  }
+
+  private final def doRunningWindowGroupedScan(
+      partByPositions: Array[Int],
+      inputCb: ColumnarBatch,
+      outputColumns: Array[ColumnVector]): Unit = {
+    val allAggs = runningWindowOptimizedData.map {
+      case (func, _) =>
+        func.groupByScan(inputCb)
+    }.toSeq
+
+    // Part by is always ascending with nulls first
+    val sortedGroupingOpts = GroupByOptions.builder()
+        .withKeysSorted(true)
+        .withKeysDescending(Array.fill(partByPositions.length)(false): _*)
+        .withKeysNullSmallest(Array.fill(partByPositions.length)(true): _*)
```

Comment:
`GroupByOptions` Javadoc states that keys default to ascending and nulls smallest if not specified, so I think this is unnecessary.